core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3797,6 +3798,16 @@ public static List distinct(List list) {
     return new ArrayList<>(result);
   }
 
+  /** Support the MAP_KEYS function. */
+  public static List mapKeys(Map map) {
+    return (List) map.keySet().stream().collect(Collectors.toList());
+  }
+
+  /** Support the MAP_VALUES function. */
+  public static List mapValues(Map map) {
+    return (List) map.values().stream().collect(Collectors.toList());
```

Comment:
`return new ArrayList<>(map.values());` 

I think it's clearer and there won't be NPE.